File: to_csv.ts
// 1. Install the Deno runtime from https://deno.land
// 2. Run this file:
// deno run -A to_csv.ts

// You may need to adjust the data directory:
const dataDirectory = "/tmp/home/lemeiz/content_refine/";

const decoder = new TextDecoder("utf-8");
const fields = ["id", "keywords", "title", "createtime"];

async function print(file: string) {
  const data = await Deno.readFile(dataDirectory + file);
  const decoded = decoder.decode(data);
  const json = JSON.parse(decoded);
  const values = fields
    .map((f) => {
      const v = json.fields.filter((e: { field: string }) => e.field === f);
      return v.length === 0 ? "" : v[0].value;
    })
    .map((v: string) => `"${v.split('"').join('""')}"`)
    .join(",");
  console.log(values);
}

console.log('"id", "keywords", "title", "createtime"');
for await (const file of Deno.readDir(dataDirectory)) {
  await print(file.name);
}

File: to_csv.py
#!/usr/bin/env python3
# -*- coding: utf8 -*-

import json
import os

import pandas as pd
from tqdm import tqdm

fields = ['id', 'keywords', 'title', 'createtime']
fields_set = set(fields)


def main(directory, out_file):
    if not out_file:
        print('No output file specified, performing dry-run')
    rows = []
    for file in tqdm(os.listdir(directory)):
        path = os.path.join(directory, file)
        with open(path) as f:
            j = json.load(f)
        rows.append({f['field']: f['value']
                     for f in j['fields']
                     if f['field'] in fields_set})
    df = pd.DataFrame(rows, columns=fields)
    if out_file:
        print('Writing to', out_file)
        df.to_csv(out_file, index=None)
    else:
        print('Dry-run, not saving the following result:')
        print(df)
    print('Success.')


if __name__ == '__main__':
    import argparse

    parser = argparse.ArgumentParser(
        formatter_class=argparse.ArgumentDefaultsHelpFormatter)

    parser.add_argument('input',
                        help='Input directory, should contain 74886 files')
    parser.add_argument('-o', '--output',
                        help='Output file, omit to perform dry-run')

    args = parser.parse_args()

    in_dir = args.input
    out_file = args.output

    main(in_dir, out_file)
